api/app/signals/apps/reporting/models/export.py

```diff
@@ -33,3 +43,66 @@ class Meta:
     isoweek = models.IntegerField()
     isoyear = models.IntegerField()
     uploaded_file = models.FileField(upload_to='exports/%Y', storage=storage)
+
+
+class CSVExportManager(models.Manager):
+    def _create_rows(self, signals_qs):
+        pass
+
+    def _dump_signals_csv(self, singals_qs, dump_dir, filename):
+        """
+        Write CSV for selected signal instances, only dump current state.
+        """
+        rows = self._create_rows(signals_qs)
+        with open(os.path.join(), 'w') as csv_file:
+            writer = csv_writer(csv_file)
+
+    def create_csv_export(self, basename, export_parameters):
+        # TODO: implement support for areas
+        interval, categories, _ = get_parameters(export_parameters)
+        categories = categories.select_related('parent')
+
+        # TODO: refactor get_parameters to use get_full_interval_info directly
+        interval = get_full_interval_info(export_parameters)
+
+        # derive zip file name
+        zip_basename = f'{basename}-{interval.desc}'
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            for cat in categories:
+                if cat.parent is None:
+                    pass  # No signals associated only with main categories
```

Comment:
`pass` or `continue` what do you want to do here? Because they behave in a different way. `continue` will go to the next iteration in the loop where `pass` does nothing and will continue with other statements

Reply:
should be `continue`